Translate Gems' section into Spanish

diegolgz opened the pull request.

diegolgz committed:
Translate 'unittesting.md' into Spanish

diegolgz committed:
Translate 'functional-programming.md' into Spanish

diegolgz committed:
Translate 'scope-guards.md' into Spanish

diegolgz committed:
Translate 'subtyping.md' into Spanish

diegolgz committed:
Translate 'compile-time-function-evaluation-ctfe.md' into Spanish

diegolgz committed:
Translate 'string-mixins.md' into Spanish

diegolgz committed:
Translate 'range-algorithms.md' into Spanish

diegolgz committed:
Translate 'uniform-function-call-syntax-ufcs.md' into Spanish

diegolgz committed:
Translate 'contract-programming.md' into Spanish

diegolgz committed:
Translate 'template-meta-programming.md' into Spanish

diegolgz committed:
Translate 'opdispatch-opapply.md' into Spanish

diegolgz committed:
Translate 'traits.md' into Spanish

diegolgz committed:
Translate 'documentation.md' into Spanish

diegolgz committed:
Translate 'bit-manipulation.md' into Spanish

diegolgz committed:
Translate 'attributes.md' into Spanish

diegolgz committed:
Translate 'unicode.md' into Spanish

diegolgz committed:
Add the index of the Gems' section

diegolgz committed:
Better translation for 'gems' ('perlas' instead of 'joyas')

csantander commented on gems/attributes.md:
```diff
@@ -0,0 +1,66 @@
+### El atributo `@property`
+
+Una función marcada con el atributo `@property` se ve como un miembro normal
+por parte de los desarrolladores:
```
Esto me parece un poco confuso. Tal vez algo así como "fuera del contexto (o entorno)"?

diegolgz replied:
No entiendo a qué te refieres aquí.

csantander replied:
El texto en inglés es "A function marked as @property looks like a normal member to the outside world." Traducirlo como "Una función marcada con el atributo `@property` se ve como un miembro normal por parte de los desarrolladores" no conlleva el mismo significado. Tampoco es muy claro a qué desarrolladores se hace referencia.

diegolgz replied:
Entiendo que "the outside world" se refiere a fuera de la propia clase, fuera del entorno, como dices, es decir, para todos los desarrolladores que usan dicha clase. Pero es verdad que es ambiguo a qué desarrolladores se refiere. Voy darle una vuelta…

csantander commented on gems/attributes.md:
```diff
@@ -0,0 +1,66 @@
+### El atributo `@nogc`
+
+Cuando el compilador de D encuentra una función marcada con el atributo
+`@nogc` se asegura que **no hay ninguna** reserva de memoria usando el
```
reservación o asignación en vez de reserva

csantander commented on gems/attributes.md:
```diff
@@ -0,0 +1,66 @@
+      }
+    }
+
+Cualquier tipo, elemento interno o elemento definido por el usuario puede ser
```
elemento interno => primitivo o básico

csantander commented on gems/contract-programming.md:
```diff
@@ -1,26 +1,29 @@
-# Contract programming
+# Programación por compromiso (*contract programming*)
```
¿Por qué no contrato en vez de compromiso?

diegolgz replied:
La verdad es que no sé cuál puede ser una buena traducción para “contract programming”, ya que ni contrato ni compromiso reflejan el verdadero significado de esa técnica. Elegí compromiso porque me sonó mejor como significado, aunque tampoco estoy convencido.

Aunque según la definición de la Wikipedia, lo lógico sí sería traducirlo por “contratos” y no por “compromisos”, aunque en español me sigue sonando mal esa traducción.

csantander commented on gems/contract-programming.md:
```diff
@@ -1,26 +1,29 @@
 
-`in` and `out` allow to formalize contracts for input
-parameters and return values of functions.
+Las palabras reservadas `in` y `out` permiten formalizar estos comprimisos
```
compromisos no está bien escrito

csantander commented on gems/documentation.md:
```diff
@@ -0,0 +1,55 @@
+tests unitarios, D permite generar [documentación](https://dlang.org/phobos/std_variant.html)
+fuera del código fuente de forma nativa.
+
+Usando un esquema estándar para documentar tipos de datosy funciones, el
```
datos y funciones (falta un espacio)

csantander commented on gems/range-algorithms.md:
```diff
@@ -1,121 +1,120 @@
+nativo y eficiente.};
 
-    // splitting predicate
+    // Predicado por donde divivir.
```
Predicado para dividir

csantander commented on gems/range-algorithms.md:
```diff
@@ -1,121 +1,120 @@
-      // of range that are chunked by the length
+      // Pone en una fila aquellos con el
+      // mismo número de caracteres. La
+      // función `chunckBy` ayuda generando
```
chunkBy

csantander reviewed (commented):
No terminé de revisar todos los cambios

diegolgz committed:
Fix some misspellings

diegolgz committed:
Use 'contrato' as 'contract' translation instead of 'compromiso'

csantander commented on gems/traits.md:
```diff
@@ -0,0 +1,180 @@
+Una de las características más potentes de D es su sistema de evaluación de
+funciones en tiempo de compilación (CTFE). Gracias a este sistema y a la
+introspección, se pueden tanto escribir programas genéricos como conseguir
+grantes optimizaciones.
```
grandes

csantander commented on gems/traits.md:
````diff
@@ -0,0 +1,180 @@
+
+```d
+auto commonPrefix(alias pred = "a == b", R1, R2)(R1 r1, R2 r2)
+  if (isForwardRange!R1
````
hace falta && al final de esta línea (la versión en inglés también necesita ser corregida)

csantander commented on gems/unicode.md:
```diff
@@ -1,80 +1,86 @@
-modern machine. D learns from the mistakes of older languages,
-as such **all** strings in D are Unicode strings, whereas strings
-in languages such as C and C++ are just arrays of bytes.
+Unicode ha solucionado muchos de esos problemás y está soportado en todos los
```
problemas

csantander commented on gems/unicode.md:
```diff
@@ -89,35 +95,34 @@ import std.range.primitives : empty,
     writeln();
 
-    // a result of auto-decoding
+    // El resulado de la decodificación
```
resultado

csantander commented on gems/unittesting.md:
```diff
@@ -0,0 +1,110 @@
+
+### Verificar ejemplos mediante `assert`
+
+Los bloques `unittest` contiene tipicamente expresiones `assert` para probar
```
típicamente

csantander commented on gems/unittesting.md:
```diff
@@ -0,0 +1,110 @@
+*/
+unittest {
+    Vector3 vec;
+    // .init a special built-in property that
```
falta traducir

diegolgz committed:
Better translation talking about 'the outside world'

In the original text, when 'to the outside world' is said,
it refers to developers that **use** the struct/class, so
that function marked with @property looks like a single
field to them.

diegolgz committed:
Fix some misspellings

diegolgz commented:
Una pregunta no relacionada con esta traducción: ¿por qué la versión en español del tour a veces se redirecciona al puerto 8080?

diegolgz commented:
Otra pregunta: ¿sería válido añadir enlaces para aclarar conceptos aunque estos no estén en la versión en inglés? Enlaces a fuentes confiables, como la Wikipedia. Por ejemplo, un enlace que creo necesario sería para aclarar qué es el _heap_, ya que no tiene una traducción al español que signifique lo mismo que en inglés (la traducción literal es _montículo_, pero no tiene ningún sentido en español hablando de informática).

csantander commented:
No sé lo del puerto 8080. Tal vez alguien más sepa? @wilzbach: Diego has noticed that the Spanish version of the Tour sometimes redirects to port 8080. Do you know why that happens?

csantander commented:
Lo de los enlaces me parece buena idea. Cómo lo hacen otros lenguajes de programación? Otra idea sería tener una sección Glosario dentro del tour.

wilzbach commented:
> Diego has noticed that the Spanish version of the Tour sometimes redirects to port 8080. Do you know why that happens?

Do you have an example or steps on how to reproduce? I just clicked myself through the spanish version and everything seems to be working fine.

diegolgz commented:
I saw that some times but I cannot reproduce now. Sorry. If I find out how I'll post it here.

wilzbach commented:
Yes, please let me/us know.
Though ideally you would file an issue against:

https://github.com/dlang-tour/core/issues

(as there are other people maintaining the DTour too)

Thanks!

wilzbach commented:
BTW don't forget to update the status of your translation here:

https://tour.dlang.org/tour/en/welcome/languages

And making an announcement to the the NG once you are done with this translation project ;-)

csantander commented:
> And making an announcement to the the NG once you are done with this translation project ;-)

I know this PR is closed, but I'm hijacking the conversation to see what @diegolagoglez and @astoliar think about announcing the translation to the NG. It should be one of you, btw, I just lurk :)

wilzbach commented:
Yes please - take credit of your hard work!

Also I think Spanish is still set on "partially translated" on tour.dlang.org ;-)

astoliar commented:
Of course, I think It's ready to announce. @diegolagoglez should do it and take credit for his great work.

diegolgz commented:
~~Ok, but one question: what is NG? The forum? Sorry for my ignorance :S~~

**Edit**: Oh, my! NG, News Group… sorry :(

I guess I have to publish this in [D:Announce](https://forum.dlang.org/group/announce), right?

csantander commented:
Yeap!

diegolgz commented:
[Done!](https://forum.dlang.org/post/aflfzwkgargwyirkimgc@forum.dlang.org)

Thanks to all who helped me with comments, corrections and PR's!
